src/React/Ref.js

```diff
@@ -0,0 +1,10 @@
+"use strict";
+
+var React = require("react");
+
+exports.createRef = React.createRef;
+
+exports.getCurrentRef_ = function(ref) {
+  if (ref.current) return ref.current;
+  else return ref;
```

Comment:
This returns an object containing `current: null` meaning in Purescript it would always be `Just` – I feel like this would be better as `else return null`.

Reply:
Great catch! If it's a DOM ref that hasn't been instantiated yet then we'll have `{ current: null }`, which will fail the first check because `null` is falsey and thus the function will return `{ current: null }` instead of just `null`.

Instead I need to check `if (ref.hasOwnProperty('current')` or similar.

Reply:
If the type of `getCurrentRef_` is `forall a. EffectFn1 (Ref a) (Nullable a)`, then `else return ref;` is ill typed, right?

Reply:
^ Discussed offline - the concern is around the `ref.current` property access potentially causing an issue if a particular ref happened to have a property of that name. To avoid this I'm going to switch the implementation so that we wrap refs in callback functions (`ref => ({ current: ref })`) and then the `.current` access in `getCurrentRef_` will always be safe.